chunker/src/main/java/edu/illinois/cs/cogcomp/chunker/main/ChunkerTrain.java

```diff
@@ -133,10 +131,9 @@ public void trainModelsWithParser(Parser parser, String modeldir, String modelna
                     cnt++;
             }
             chunker.doneWithRound();
-            writeModelsToDisk(modeldir,modelname);
             // Test on dev set
             BIOTester tester =
-                    new BIOTester(new Chunker(lcpath,lexpath), new ChunkLabel(), new ChildrenFromVectors(parser));
+                    new BIOTester(chunker, chunker.getLabeler(), new ChildrenFromVectors(parser));
```

Comment:
I assume you have you tested this bit of code? Sometimes the re-initialization was needed to reset the feature weights for testing.